### ISSUE TITLE
Deploy Issues

### ISSUE THREAD

**Issue by cmargar** [firelab/FWAS-web#26]
Hi all,

recently I've tried to deploy the project in my local computer to test the service and got some troubles in my way. I've tried with and without the docker images. Let me give us a list of issues I've found:

**Docker system:** 
- postgreSQL docker sometimes breaks and needs to be restarted after deploying (guess caused by creating extensions outside the gis table?).

**For Fetchers**
- Currently only haves HRRR fetcher implemented on the cron jobs along the check alert one.

- Sometimes the gdalwarp conversion fails randomly in a grib2 saying it's an invalid file fromat and breaks the worker (reinstalling GDAL again looks like solved it). We should know which gdal version and dependencies shall we need to run it correctly.

- After a raster has been converted to PostGIS and system tries to insert each sql file generated a couple of things happen: cpu usage increases drastically, with each raster inserted on DB the next ones takes more and more time. Maybe caused by “ANALYZE / VACUUM” commands (not sure about this but this makes impossible to make a cron job with more fetchers. Last test took like an hour to download, convert, generate sql and insert them.
Timeout on the job had to be increased to be able to 

- After the insert of each raster image, an update with dates and source fields fails (guess the Database object looses connection after creating a new fetcher object).

**For API:**
- Create alert not working

In general lines, looks like the project in the master branch (the one I've used) it's in the middle of a model/logic migration (flask to fastAPI, fetchers redo logic and overall review of all code). And there is an "old" folder with code that I guess was working in older versions before these changes,

I have been solving errors as they arose to try to deploy the project (without dockers, fixing create alert, fetcher run with a new database object, running the migration, updaating postgis/gdal...) in my local environment to a point where it is already impossible to continue.

Tell me if you need more information

**Comment by ldmalott** [comment]
@cmargar thanks for reporting. I went through and handled the initialization and create alert issues on https://github.com/firelab/FWAS/commit/3d15f58e87f0d9e02433d1667d0b45f1f6d9f35b. I'm able to spin up via `docker-compose up -d` and get the service running. API docs are shown at http://localhost:8000/docs. 

The `old` folder was all the code prior to turning this repo into a service. I moved it there since it still contains useful references for some of the geospatial transforms and data integrations. 

For the CPU utilization, that part is at least expected. What isn't expected is the increase in fetch/transform/save duration over time. Could be from something that is poorly indexed or updating records when it shouldn't. I suspect there is a full-table scan or full-table update that shouldn't be happening. I can do some more testing there.

For the variety of fetchers, only HRRR is implemented in a stable state. I've played around with some prototypes for others but haven't merged them.

**Comment by cmargar** [comment]
@ldmalott Hello again! Thanks for your fast response. I have some new about the changes:

- Docker
I can not make it to work with the ```docker-compose up -d```. It downloads everything and tries to initialize them but postgres one fails and then also fails the backend because there isn't a running database on the db alias container. I attached you a screenshot of the logs in the container. Looks like the pg_cron extensions breaks everything and then the container restarts itself in a loop due to the docker config.
![Screenshot from 2020-06-10 08-49-16](https://user-images.githubusercontent.com/66615311/84244902-95d1e080-ab04-11ea-806f-2895db6a7b0b.png)

![Screenshot from 2020-06-10 08-54-25](https://user-images.githubusercontent.com/66615311/84242967-d7ad5780-ab01-11ea-9c15-f90bb2816e7c.png)

So I tried without them again, creating a database and tweaking the Redis and postgreSQL variables.

- dotenv variables
All services are specified here so first I've changed the redis and postgre to my localhost services and credentials. For some reason couldn't make it to work so I replaced the strings where needed.

- HRRR fetcher
On the run_now function, couldn't make it to work with the hardcoded connection string or the SQLALCHEMY_DATABASE_URI variable. It throws ```with Database(SQLALCHEMY_DATABASE_URI) as db:
AttributeError: __enter__```
For the CPU issues yeah it's totally fine I guess but the increasing time will be a trouble in the long term. Noticed that even doing a TRUNCATE on the table and restarting the identity, next ID's are still growing up.

- Fetchers
Yeah, I've noticed that, the old logic was there and will take some time to migrate to new model.

- API
As you said the create alert now works after the migration model, yay!.

- Notification System
Guess it was working time ago with the old code, saw some parts where the service sent emails/SMS to users but I think it's not implemented now. Are you planning to include it or we need to implement our fetch/send created notifications?

---------------------------

So, still need some help with this docker setup or local configuration without them. This time just couldn't make it to process a single fetch. Still need to check the notification creation after a check alert job.
In a production environment I guess will be using containers to not mess up with our local redis/postgres and changing the docker ports to listen.

Thank you for your time. By the way I'm an employee of Technosylva and this service will be included in a current project.
